src/providers/go-to-definition.ts

```diff
@@ -0,0 +1,23 @@
+import * as vscode from "vscode";
+
+export class MF6DefinitionProvider implements vscode.DefinitionProvider {
+  public async provideDefinition(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+  ): Promise<vscode.Location | vscode.Location[]> {
```

Comment:
```suggestion
  ) {
```

When implementing an interface, the linter knows about the return type of its components. In this instance, it knows the return type of `provideDefinition`. Thus, it's better to remove the return type so that the linter can help when you're defining the logic inside the method. Try returning `1`, it will complain that the return value doesn't match the required return type of the method.

As for the type of the arguments, I tried removing but it looks like you are required to specify it.